src/node/mod.rs

```diff
@@ -418,6 +418,15 @@ impl Node {
             .ok_or(RoutingError::InvalidState)
     }
 
+    /// Returns our index in the current BLS group or `RoutingError::InvalidState` if we are not
+    /// elder.
```

Comment:
better change the comment to `if section key was not generated yet.`
the `approved` function also includes the stage of `adult`